src/modules/core/components/CountDownTimer/CountDownTimer.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage, defineMessage } from 'react-intl';
+
+import { MiniSpinnerLoader } from '~core/Preloaders';
+
+import { useVotingExtensionParamsQuery } from '~data/index';
+import { Address } from '~types/index';
+
+import styles from './CountDownTimer.css';
+import { calculateTimeLeft } from './calculateTimeLeft';
+
+const MSG = defineMessage({
+  stake: {
+    id: 'CountDownTimer.CountDownTimer.stake',
+    defaultMessage: 'Time left to stake',
+  },
+  motionPass: {
+    id: 'CountDownTimer.CountDownTimer.motionPass',
+    defaultMessage: 'Motion will pass in',
+  },
+  motionFail: {
+    id: 'CountDownTimer.CountDownTimer.motionFail',
+    defaultMessage: 'Motion will fail in',
+  },
+  reveal: {
+    id: 'CountDownTimer.CountDownTimer.reveal',
+    defaultMessage: 'Reveal ends in',
+  },
+  voting: {
+    id: 'CountDownTimer.CountDownTimer.voting',
+    defaultMessage: 'Voting ends in',
+  },
+  days: {
+    id: 'CountDownTimer.CountDownTimer.days',
+    defaultMessage: ' {days}d',
+  },
+  hours: {
+    id: 'CountDownTimer.CountDownTimer.hours',
+    defaultMessage: ' {hours}h',
+  },
+  minutes: {
+    id: 'CountDownTimer.CountDownTimer.minutes',
+    defaultMessage: ' {minutes}m',
+  },
+  seconds: {
+    id: 'CountDownTimer.CountDownTimer.seconds',
+    defaultMessage: ' {seconds}s',
+  },
+  loadingText: {
+    id: 'CountDownTimer.CountDownTimer.loadingText',
+    defaultMessage: 'Loading countdown period',
+  },
+});
+
+interface Props {
+  copyOption: 'stake' | 'motionPass' | 'motionFail' | 'reveal' | 'voting';
+  periodType:
+    | 'stakePeriod'
+    | 'submitPeriod'
+    | 'revealPeriod'
+    | 'escalationPeriod';
+  createdAt: number;
+  colonyAddress: Address;
+}
+
+const CountDownTimer = ({
+  copyOption,
+  colonyAddress,
+  createdAt,
+  periodType,
+}: Props) => {
+  const { data, error, loading } = useVotingExtensionParamsQuery({
+    variables: { colonyAddress },
+  });
+
+  const stakePeriod = data?.votingExtensionParams[periodType];
+
+  const [timeLeft, setTimeLeft] = useState(
+    calculateTimeLeft(createdAt, stakePeriod),
+  );
+
+  useEffect(() => {
+    if (stakePeriod !== undefined) {
+      const timer = setInterval(() => {
+        setTimeLeft(calculateTimeLeft(createdAt, stakePeriod));
+      }, 1000);
+      return () => clearInterval(timer);
+    }
+    return undefined;
+  }, [createdAt, stakePeriod]);
+
+  if (loading) {
+    return <MiniSpinnerLoader loadingText={MSG.loadingText} />;
+  }
```

Comment:
You need to include the `data === undefined` check in here as well.

Otherwise, you get a brief flash of the loading state, then it disappears until it loads all the data, then finally it shows the time.

Here's the demo:

![bug-countdown-spinner](https://user-images.githubusercontent.com/1193222/113732379-37649700-9702-11eb-9327-2573f4c974a0.gif)